content/blog/2025-06-30-cancellation.md

```diff
@@ -0,0 +1,491 @@
+---
+layout: post
+title: The Challenge of Cancelling Long-Running Queries
```

Comment:
I would like to suggest an alternate title and I didn't want to make the change without asking first

I think this article is super valuable both for the in-depth discussion on cancelling but also for the background on execution using Streams . I think highlighting this in the title might help more people find and want to read it

What do you think?

```suggestion
title: Using Rust async for Execution and Cancelling Long-Running Queries
```